UI 1차 완성 버전 master 반영

HyunsangHan opened the pull request:
해커톤 이후, 추가 작업을 통해 미완성이었던 UI가 거의 갖추어졌습니다.
한번 `master` 브랜치 업데이트를 진행하고자 합니다.

Closes #10 
Closes #60 
Closes #62 
Closes #63 
Closes #66 
Closes #68 
Closes #76

HyunsangHan committed:
Header컴포넌트 props 리팩토링, Profile css리팩토링, JoinButton resolves #46

c-yeonwoo committed:
미팅그룹생성 투탭 css 개선

c-yeonwoo committed:
새로운 그룹 생성시 이미지 3개를 다 채워야 저장됨

c-yeonwoo committed:
미팅 그룹 생성 two tab의 디폴트값 => 지난 그룹

HyunsangHan committed:
TeamProfile 기본 틀 css 수정

HyunsangHan committed:
Profile.js, TeamProfile.js 디자인 반영하여 리팩토링

HyunsangHan committed:
Merge pull request #53 from HyunSangHan/feature/player

Feature/player

HyunsangHan committed:
pull develop branch and merge

HyunsangHan committed:
TeamProfile 문구 수정

HyunsangHan committed:
TeamProfile 컴포넌트 구조 개선

HyunsangHan committed:
TeamProfileBody 리팩토링 - 불필요한 props 제거

HyunsangHan committed:
TeamProfile에서 팀사진 전체너비로 스와이프 가능한 UI 적용

HyunsangHan committed:
불필요 css제거, 800px 이상 시 first-child의 ml 수정

HyunsangHan committed:
flex 관련 cross-browsing

HyunsangHan committed:
TeamProfile.js 이미지영역 디자인 적합성 고도화(자연스러운 인터랙션, 작은 width에서 이미지영역 크기 작아지게 수정, NEW탭에서 안내문구 추가

HyunsangHan committed:
Merge pull request #54 from HyunSangHan/feature/refactoring

Feature/refactoring

c-yeonwoo committed:
'

c-yeonwoo committed:
Merge branch 'develop' of https://github.com/HyunSangHan/MeetingTime into feature/player

c-yeonwoo committed:
main.js counter profile에  image-wrapper 추가

c-yeonwoo committed:
window 축소시 scroll focus at first image

c-yeonwoo committed:
안주쏘기 팝업 구현 (prop 내리는 걸로 수정, 및 main.js로 이동, css 수정)

c-yeonwoo committed:
그룹생성시 team popup css 구현 (but 저장 오류 해결 x)

c-yeonwoo committed:
gift-popup 다시 controltool.js 로 이동

HyunsangHan committed:
카카오 로그인 버튼 커스터마이징 - Resolves #52

c-yeonwoo committed:
기존 그룹에 기존데이터 렌더링

c-yeonwoo committed:
그룹생성 버튼 분기 (이미지 세장 올려야 버튼활성화)

c-yeonwoo committed:
git popup button 버그 수정 & 콜?/콜!! 표기 반대로 수정

HyunsangHan committed:
TeamProfileBody에 있던 오타수정, 불필요한 state제거, views.py에서 img_modified_at이 수정되도록 변경

HyunsangHan committed:
MeetingInfo에 데이터 꿰매기

HyunsangHan committed:
current_meeting이 뽑히지 않는 경우 맨 마지막 미팅을 가져오도록 예외처리 수정

HyunsangHan committed:
D-day계산 잘못된 코드 수정

HyunsangHan committed:
d-day(getDateGap)함수 로직 수정

HyunsangHan committed:
CountDown 개발(단, 브라우저가 멈추면 멈춘 시간이 반영되지 않아 추후 개선필요

HyunsangHan committed:
count down 컴포넌트 폰트 수정

HyunsangHan committed:
Fragment 적용

HyunsangHan committed:
Resolves #47

HyunsangHan committed:
count_down 문구 로직 함수화

HyunsangHan committed:
count_down 내 코드 리팩토링(기능 자체는 그대로 유지)

c-yeonwoo committed:
Merge pull request #55 from HyunSangHan/feature/player

Feature/player

HyunsangHan committed:
develop branch와의 conflict해결, 불필요한 이미지파일 staged에서 제외

HyunsangHan committed:
conflict해결, secondGap 을 let으로 수정

HyunsangHan committed:
Merge pull request #56 from HyunSangHan/feature/refactoring

Feature/refactoring

c-yeonwoo committed:
태그 및 세미콜론 약간 리팩토링

c-yeonwoo committed:
main profile images-wrapper 수정

c-yeonwoo committed:
controlTool.js 변수명 수정

c-yeonwoo committed:
conflict해결

HyunsangHan committed:
timeNotification 함수 내 return 위치 변경

notification 변수 없애고 if문 안에서 바로 return하는 것으로 수정

HyunsangHan committed:
Merge pull request #57 from HyunSangHan/feature/count_down

Feature/count down

c-yeonwoo committed:
main image-wrapper 재수정

c-yeonwoo committed:
is_gift_male_on 수정

c-yeonwoo committed:
conflict 해결 but current_matching 조회 fail

c-yeonwoo committed:
main.js image-wrapper 및 기타 마진 수정

c-yeonwoo committed:
total-container width 100% 수정

HyunsangHan committed:
Merge pull request #58 from HyunSangHan/feature/player

Feature/player

HyunsangHan committed:
scss 겹친 class명 수정하여 TeamProfile.js 디자인 복구

HyunsangHan committed:
Main.js 관련 스타일링 리팩토링

HyunsangHan committed:
잘못 구현된 UI들 수정 Closes #61

HyunsangHan committed:
Rename profile.scss to Profile.scss

HyunsangHan committed:
Profile.js에 OO대 형태로 옵션 수정, Profile.scss 파일명 대문자 변환

HyunsangHan committed:
Profile.scss 이름 해결 준비

HyunsangHan committed:
Profile.scss 이름 해결

HyunsangHan committed:
Merge branch 'develop' into feature/refactoring

HyunsangHan committed:
Merge pull request #59 from HyunSangHan/feature/refactoring

Feature/refactoring

HyunsangHan committed:
ToolTip 컴포넌트의 플랫폼화(자유도 증가)

HyunsangHan committed:
ToolTip 컴포넌트의 플랫폼화(자유도 증가)

HyunsangHan committed:
ToolTipDown 도입, css 수정

HyunsangHan committed:
outdated 삭제

HyunsangHan committed:
테스트용 주석 복구

HyunsangHan committed:
불필요 파일 삭제

HyunsangHan committed:
프로필 버튼 위치 수정

HyunsangHan committed:
Closes #63 issue - 버튼 위치 조정 및 중복 문자 제거

HyunsangHan committed:
Merge branch 'feature/refactoring' of https://github.com/HyunSangHan/MeetingTime into feature/tooltip

HyunsangHan committed:
ToolTipDown UI 완성

HyunsangHan committed:
안주쏘기를 했을 때 상대방 화면에 툴팁 나타나도록 구현

HyunsangHan committed:
Merge pull request #65 from HyunSangHan/feature/refactoring

Feature/refactoring

dependabot[bot] committed:
Bump lodash.template from 4.4.0 to 4.5.0 in /frontend

Bumps [lodash.template](https://github.com/lodash/lodash) from 4.4.0 to 4.5.0.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.4.0...4.5.0)

Signed-off-by: dependabot[bot] <support@github.com>

HyunsangHan committed:
Add READMD.md + Merge pull request #69 from HyunSangHan/dependabot/npm_and_yarn/frontend/lodash.template-4.5.0

Bump lodash.template from 4.4.0 to 4.5.0 in /frontend

dependabot[bot] committed:
Bump django from 2.2.1 to 2.2.4 in /backend

Bumps [django](https://github.com/django/django) from 2.2.1 to 2.2.4.
- [Release notes](https://github.com/django/django/releases)
- [Commits](https://github.com/django/django/compare/2.2.1...2.2.4)

Signed-off-by: dependabot[bot] <support@github.com>

HyunsangHan committed:
Merge pull request #70 from HyunSangHan/dependabot/pip/backend/django-2.2.4

Bump django from 2.2.1 to 2.2.4 in /backend

HyunsangHan committed:
Update dependencies
security issue가 있는 dependency들이 있어 일괄 업데이트 했음. 테스트 결과 이상없이 serving되고 있어 push함.

HyunsangHan committed:
백엔드 의존성 업데이트

tyj9327 commented on frontend/src/components/Initpage.js:
```diff
@@ -25,39 +24,50 @@ class Initpage extends Component {
+                })
+                .catch( err => {
+                    console.log(err);
+                });
```
사소한 것이지만 axios를 여러 개 겹쳐서 사용하면 error catch는 가장 바깥에 한 번만 해도 안에 것까지 다 catch 된다고 하네요!

HyunsangHan replied:
헐 저도 이제야 이런 게 보이네요..
axios는 프로미스 기반이라서 일단 이렇게 콜백지옥스럽게 깊어질 필요가 없이 then으로 계속 체이닝해서 쓰면 되겠고,
에러객체는 캐치해주지 않으면 계속 위로 에러를 던지는(throw) 형태로 작동하므로 캐치는 한번만 써주면 되겠습니다.
리팩토링 할 것이 엄청 많겠어요..!

tyj9327 commented:
양이 너무 많고 제가 캐치업하기에 프론트를 처음부터 이해해야 할 것 같아서 학기가 끝난 후에 제대로 리뷰가 가능할 것 같네요 ㅠㅠ

HyunSangHan commented:
> 양이 너무 많고 제가 캐치업하기에 프론트를 처음부터 이해해야 할 것 같아서 학기가 끝난 후에 제대로 리뷰가 가능할 것 같네요 ㅠㅠ

큰 문제가 없다면 현 버전대로 merge하고 종강이후에 한번 날 잡고 전체적으로 리팩토링하는 게 어떨까 싶습니다.
(이로써 예상보다 +1 day 정도 업무량이 늘었습니다.)

tyj9327 reviewed (approved):
그렇다면 큰 문제가 없기를 바라며!

HyunsangHan committed:
Merge pull request #73 from HyunSangHan/hotfix

백엔드 의존성 업데이트

HyunSangHan commented:
@tyj9327 @qwerty98765 @choiyeonseok 
master <- develop 으로의 PR은, PR과정에서 실수 없는지 등 부수적인 부분만 리뷰하고, 실질적인 코드리뷰는 develop <- feature/XXX 으로의 PR에서 진행하면 될 것 같습니다. 그런 의미에서 #67 에 대한 코드리뷰를 해주시면 좋을 듯합니다.

HyunsangHan committed:
Add polyfill for ie

HyunsangHan committed:
Install Prettier as a dependency

HyunsangHan committed:
Apply prettier all in src directory

HyunsangHan committed:
Install djangorestframework-camel-case

HyunsangHan committed:
Unify unnecessary snake_case to camelCase
1. API Reponse가 카멜케이스로 자동 변환되어 오도록 라이브러리 적용
2. Redux의 store에서도 state 관리를 카멜케이스로 변경
Closes #76

HyunsangHan committed:
Resolve warning messages

HyunsangHan committed:
Make App.js only manage  Route

HyunsangHan committed:
Use Loading component by isStoreLoaded

HyunsangHan committed:
Fix redirect conditions

HyunsangHan committed:
Delete duplicated catch

HyunsangHan committed:
Make loading component spin

HyunsangHan committed:
Apply object property value shorthand

HyunsangHan committed:
Fix name of action

HyunsangHan committed:
Add catch err

HyunsangHan committed:
Implement HOC

HyunsangHan committed:
Delete redux codes from JoinButton

HyunsangHan committed:
Delete redux codes from ControlTool

HyunsangHan committed:
Add clearInterval when unmounting CountDown

HyunsangHan committed:
Fix syntax of '.then()'

HyunsangHan committed:
Add ERD in README

HyunsangHan committed:
Merge pull request #67 from HyunSangHan/feature/tooltip

Main 컴포넌트에 TooTipDown 적용(안주쏘기 받았을 때의 말풍선)

HyunsangHan committed:
Add recognizing logic for init

HyunsangHan committed:
Fix CALL button not working issue of control tool

HyunsangHan committed:
Remove unnecessary popups by using window.alert or confirm

HyunsangHan committed:
Fix bug about ControlTool button

HyunsangHan committed:
Merge pull request #78 from HyunSangHan/refactoring

전반 리팩토링
